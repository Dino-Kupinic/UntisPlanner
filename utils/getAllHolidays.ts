import type {AttributeConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"
import Holidays from "date-holidays"
const hd: Holidays = new Holidays("AT")

const excludedHolidays: string[] = ["10-26"]
export default function (year: number, language: string = "en"): AttributeConfig[] {
  const h = hd.getHolidays(year, language)

  const returnArray: AttributeConfig[] = []
  for (const holiday of h) {
    if (excludedHolidays.includes(holiday.date)) {
      continue
    }
    const temp: AttributeConfig = {
      key: holiday.name,
      highlight: "red",
      dates: [{
        start: new Date(holiday.start),
        end: new Date(holiday.start),
      }],
      popover: {
        label: holiday.name,
        visibility: "hover",
      },
    }
    returnArray.push(temp)
  }
  return returnArray
}