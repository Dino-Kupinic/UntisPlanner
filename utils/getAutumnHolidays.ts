import type {AttributeConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"

export default function (year: number): AttributeConfig {
  return {
    key: "Autumn holidays",
    highlight: "teal",
    dates: [{
      start: new Date(year, 9, 27),
      end: new Date(year, 9, 31),
    }],
    popover: {
      label: "Autumn holidays",
      visibility: "hover",
    },
  }
}