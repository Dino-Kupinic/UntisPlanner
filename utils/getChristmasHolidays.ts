import type {AttributeConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"
import {MINIMUM_YEAR} from "~/model/constants"

export default function (year: number = MINIMUM_YEAR): AttributeConfig {
  const holidayStart: Date = new Date(year, 11, 24)
  const holidayEnd: Date = new Date(holidayStart)
  holidayEnd.setDate(holidayStart.getDate() + 14)

  return {
    key: "Christmas holidays",
    highlight: "indigo",
    dates: [{
      start: holidayStart,
      end: holidayEnd,
    }],
    popover: {
      label: "Christmas holidays",
      visibility: "hover",
    },
  }
}