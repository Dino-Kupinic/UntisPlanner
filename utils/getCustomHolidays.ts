import type {AttributeConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"
import type {ExportHoliday} from "~/model/holiday"

/**
 * Utility function to create an Attribute objects for the custom holidays
 * @param holidays - The holidays to create the attributes for
 */
export default function (holidays: ExportHoliday[]): AttributeConfig[] {
  const returnArray: AttributeConfig[] = [{}]
  // TODO: add the color to the type?
  const COLOR: string = "red"

  holidays.forEach((item: ExportHoliday) => {
    const temp: AttributeConfig = {
      key: item.name,
      highlight: COLOR,
      dates: [{
        start: new Date(item.start),
        end: new Date(item.end),
      }],
      popover: getPopOver(item.name),
    }
    returnArray.push(temp)
  })
  return returnArray
}