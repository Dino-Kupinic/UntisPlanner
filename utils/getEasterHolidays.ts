import type {AttributeConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"
// @ts-ignore
import {easter} from "date-easter"
import Holidays from "date-holidays"

/**
 * The holidays class which manages the holidays
 */
const hd: Holidays = new Holidays("AT")

/**
 * Utility function to get the Easter holidays
 * @param year - The year to get the holidays for
 * @param language - The language to get the holidays for
 */
export default function (year: number, language: string = "de-at"): AttributeConfig {
  const holidays = hd.getHolidays(year, language)

  const easterDay = easter(year);
  const palmSaturday = new Date(easterDay);
  palmSaturday.setDate(palmSaturday.getDate() - 8)

  const easterMonday = holidays.filter((holiday) => holiday.name === "Ostermontag")[0].date;

  return {
    key: "Easter holidays",
    highlight: "pink",
    dates: [{
      start: palmSaturday,
      end: easterMonday,
    }],
    popover: {
      label: "Easter holidays",
      visibility: "hover",
    },
  }
}
