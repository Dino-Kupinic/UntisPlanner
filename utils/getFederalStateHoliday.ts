// import type {AttributeConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"
// import {MAXIMUM_YEAR, MINIMUM_YEAR} from "~/model/constants"
//
// /**
//  * The color for the holiday
//  */
// const DEFAULT_HOLIDAY_COLOR: string = "purple"
//
// /**
//  * Checks if the color isn't empty
//  * @param holidayColor
//  */
// function isValidColor(holidayColor: string): boolean {
//   return holidayColor.trim() !== ""
// }
//
// /**
//  * Utility function to get the federal state holiday
//  * @param holidayColor - The color for the holiday
//  * @returns AttributeConfig
//  */
// // TODO: if needed, add option for providing highlight object
// export default function (holidayColor: string = DEFAULT_HOLIDAY_COLOR): AttributeConfig {
//   const MONTH_OCTOBER: number = 10
//   const NATIONAL_HOLIDAY_DAY: number = 26
//
//   if (!isValidColor(holidayColor)) {
//     throw new Error("Invalid color")
//   }
//
//   return {
//     key: "nationalHoliday",
//     highlight: holidayColor,
//     dates: [{
//       repeat: {
//         every: [12, "months"],
//         from: new Date(MINIMUM_YEAR, MONTH_OCTOBER - 1, NATIONAL_HOLIDAY_DAY),
//         until: new Date(MAXIMUM_YEAR, MONTH_OCTOBER - 1, NATIONAL_HOLIDAY_DAY),
//         on: dayParts => dayParts.day == NATIONAL_HOLIDAY_DAY && dayParts.month == MONTH_OCTOBER,
//       },
//     }],
//     popover: getPopOver("National holiday"),
//   }
// }
