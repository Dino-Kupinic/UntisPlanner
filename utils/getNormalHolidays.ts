import type {AttributeConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"
import Holidays from "date-holidays"

/**
 * The holidays class which manages the holidays
 */
const hd: Holidays = new Holidays("AT")
/**
 * The holidays to exclude
 */
// TODO: Go from string check to date object check
const excludedHolidays: string[] = ["10-26"]
/**
 * Utility function to get the normal holidays of a year
 * @param year - The year to get the holidays for
 * @param language - The language to get the holidays for
 * @param color - The color for the holidays
 * @returns AttributeConfig[]
 */
export default function (year: number, language: string = "en", color: string = "red"): AttributeConfig[] {
  const holidayList = hd.getHolidays(year, language)

  const returnArray: AttributeConfig[] = []
  for (const holiday of holidayList) {
    if (excludedHolidays.includes(holiday.date)) {
      continue
    }
    const temp: AttributeConfig = {
      key: holiday.name,
      highlight: color,
      dates: [{
        start: new Date(holiday.start),
        end: new Date(holiday.start),
      }],
      popover: {
        label: holiday.name,
        visibility: "hover",
      },
    }
    returnArray.push(temp)
  }
  return returnArray
}