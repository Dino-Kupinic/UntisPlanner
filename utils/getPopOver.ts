import type {PopoverConfig} from "v-calendar/dist/types/src/utils/attribute.d.ts"

/**
 * The visibility options for the popover
 */
type VisibilityOptions = "click" | "hover" | "hover-focus" | "focus"
/**
 * Utility function to create a PopoverConfig object
 *
 * @param label - The label to display in the popover
 * @param visibility - The visibility of the popover. Default is "hover"
 * @returns PopoverConfig
 */
export default function (label: string, visibility: VisibilityOptions = "hover"): PopoverConfig {
  return {
    label: label,
    visibility: visibility,
  }
}