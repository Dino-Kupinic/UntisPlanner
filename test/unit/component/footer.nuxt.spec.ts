import {it, expect} from "vitest"
import {mountSuspended} from "vitest-environment-nuxt/utils"
import {Footer} from "#components"

it("Check footer display", async () =>{
  const component = await mountSuspended(Footer)
  expect(component.findComponent("GithubIcon")).toBeTruthy()
  expect(component.findComponent("HeartIcon")).toBeTruthy()
})