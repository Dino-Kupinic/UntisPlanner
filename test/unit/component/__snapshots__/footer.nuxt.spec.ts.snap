// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > Display correct license text 1`] = `
"<footer><br>
  <div class="flex items-center max-sm:space-y-3 max-sm:flex-col max-sm:flex-wrap max-sm:content-center max-sm:items-center border-t border-neutral-300 dark:bg-gray-900 dark:border-gray-800 pt-3 pb-5">
    <div class="basis-1/3"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" class="dark:fill-white fill-black m-auto" width="30" height="30" viewBox="0 0 16 16"><a href="https://github.com/Dino-Kupinic/UntisPlanner" rel="noopener noreferrer" target="_blank">
          <path fill="currentColor" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59c.4.07.55-.17.55-.38c0-.19-.01-.82-.01-1.49c-2.01.37-2.53-.49-2.69-.94c-.09-.23-.48-.94-.82-1.13c-.28-.15-.68-.52-.01-.53c.63-.01 1.08.58 1.23.82c.72 1.21 1.87.87 2.33.66c.07-.52.28-.87.51-1.07c-1.78-.2-3.64-.89-3.64-3.95c0-.87.31-1.59.82-2.15c-.08-.2-.36-1.02.08-2.12c0 0 .67-.21 2.2.82c.64-.18 1.32-.27 2-.27c.68 0 1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82c.44 1.1.16 1.92.08 2.12c.51.56.82 1.27.82 2.15c0 3.07-1.87 3.75-3.65 3.95c.29.25.54.73.54 1.48c0 1.07-.01 1.93-.01 2.2c0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8"></path>
        </a></svg></div>
    <div class="basis-1/3 text-center">
      <p>Made with <svg xmlns="http://www.w3.org/2000/svg" width="20px" height="20px" viewBox="0 0 24 24" class="inline-block pb-0.5">
          <path fill="red" fill-opacity="0" d="M12 20L20.5 11V7L17 5.5L12 7L7 5.5L3.5 7V11L12 20Z">
            <animate fill="freeze" attributeName="fill-opacity" begin="0.5s" dur="0.5s" values="0;1"></animate>
          </path>
          <path fill="none" stroke="red" stroke-dasharray="30" stroke-dashoffset="30" stroke-linecap="round" stroke-width="2" d="M12 8C12 8 12 8 12.7578 7C13.6343 5.84335 14.9398 5 16.5 5C18.9853 5 21 7.01472 21 9.5C21 10.4251 20.7209 11.285 20.2422 12C19.435 13.206 12 21 12 21M12 8C12 8 12 8 11.2422 7C10.3657 5.84335 9.06021 5 7.5 5C5.01472 5 3 7.01472 3 9.5C3 10.4251 3.27914 11.285 3.75777 12C4.56504 13.206 12 21 12 21">
            <animate fill="freeze" attributeName="stroke-dashoffset" dur="0.5s" values="30;0"></animate>
          </path>
        </svg></p>
    </div>
    <div class="basis-1/3 text-sm text-center">
      <p>GNU General Public License v3.0</p>
      <p>v0.2.2</p>
    </div>
  </div>
</footer>"
`;
